test/controllers/actions/ChannelActionSpec.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.actions
+
+import controllers.routes
+import models.ChannelType.{api, web}
+import models.requests.ChannelRequest
+import org.scalatest.EitherValues
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import play.api.http.Status
+import play.api.http.Status.BAD_REQUEST
+import play.api.mvc.{Headers, Request, Result}
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{defaultAwaitTimeout, status}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.xml.NodeSeq
+
+class ChannelActionSpec extends AnyFreeSpec with Matchers with ScalaFutures with EitherValues {
+
+  private class Harness extends ChannelAction {
+    def call[A](request: Request[A]): Future[Either[Result, ChannelRequest[A]]] =
+      refine(request)
+  }
+
+  "ChannelAction" - {
+    "returns BadRequest if the channel header is missing" in {
+      val futureResult = new Harness().call(FakeRequest("POST", routes.MessagesController.post().url))
+
+      whenReady(futureResult) {
+        result =>
+          result mustBe a[Left[Status, _]]
+
+        status(Future.successful(result.left.get)) mustBe BAD_REQUEST
```

Comment:
We have EitherValues mixed in and should make use of that here for better test errors. We can also get the status using: `result.left.value.header.status mustBe BAD_REQUEST` (similar comments for below)